### ISSUE TITLE
Don't hoist variables in the Pallene->Lua translator

### ISSUE THREAD

**Issue by hugomg** [pallene-lang/pallene#367]
As part of the changes we have been making to the Toplevel, the Pallene->Lua translator should stop hoisting the toplevel variables. For example, this program:

```
local m: module = {}
local xs: integer = 10
return m
```

Currently is translated to this program after running `pallenec --emit-lua`

```
local m, xs;m = {}
xs = 10
return m
```

We should instead produce

```
local m = {}
local xs = 10
return m
```

The relevant test cases are in the translator_spec

**Comment by hugomg** [comment]
We might also need to make changes to the lexer and parser. The translator works by erasing the sections of the program that have type annotations and it is the responsibility of the lexer and parser to annotate these sections.

**Comment by hugomg** [comment]
Closed by #374
